docs/jwst/skymatch/description.rst

```diff
@@ -0,0 +1,239 @@
+described above, for a set of 6 input images, each with different
+sky signal levels. The first column gives the actual sky values imposed into
+the images, with the subsequent columns showing the results from each method.
+
++-------+-------+--------+-------+-------+--------------+
+| Image | Local | Global | Match | Match | Global+Match |
+|       |       |        |       |       |              |
+| Value |       |        | Down  | Up    | Down         |
++=======+=======+========+=======+=======+==============+
+| 200   |  200  |  200   |    0  |  -300 |        200   |
++-------+-------+--------+-------+-------+--------------+
+| 300   |  300  |  200   |  100  |  -200 |        300   |
++-------+-------+--------+-------+-------+--------------+
+| 350   |  350  |  200   |  150  |  -150 |        350   |
++-------+-------+--------+-------+-------+--------------+
+| 400   |  400  |  200   |  200  |  -100 |        400   |
++-------+-------+--------+-------+-------+--------------+
+| 500   |  500  |  200   |  300  |     0 |        500   |
++-------+-------+--------+-------+-------+--------------+
+| 370   |  370  |  200   |  170  |  -130 |        370   |
++-------+-------+--------+-------+-------+--------------+
+
+- "local" just finds the sky level of each image independently of the rest.
+- "global" finds the minimum sky level amongst all images.
+- "match" with "match_down=True" finds the offset needed to match all images
+  to the level of the image with the lowest sky level.
+- "match" with "match_down=False" finds the offset needed to match all images
+  to the level of the image with the highest sky level.
+- "global+match" with "match_down=True" finds the offsets and global value
+  needed to set all images to a sky level of zero.
+
+In methods that find sky background levels in each image ("local") or
+a single level for all images ("global"), image statistics are usually
+computed using sigma clipping. If the input images contain vast swaths of empty
+sky, then the sigma clipping algorithm should be able to automatically
+exclude (clip) contributions from bright compact sources.
+In this case the measured sky background is the
+measured signal level from the empty sky regions. On the other hand, the
+"match" method compares the *total* signal levels integrated over the regions
+in the images that correspond to common (overlap) regions on the celestial
+sphere for both images being compared (comparison is pair-wise).
+This method is often used when there are no large
+empty sky regions in the images, such as when a diffuse extended source occupies
+most of the view. This method cannot measure true background, but
+rather additive corrections that need to be applied to the input images so that
+the total signal from the same part of the sky is equal in all images.
+
+The computed sky value for each image is stored in the BKGLEVEL keyword of the
+output data products. The BKGSUB keyword (boolean) is also populated to indicate
```

Comment:
Are the values in the table above the ones that are stored in the BKGLEVEL keyword? If so, this behavior makes me a bit uneasy - the step output seems very modal in whether it's giving you the actual estimate of the background (for local or global), a value to match other exposures (match_down or match_up), or an estimate of the background like the first, but instead of the background value, a number to be removed from the image to reach a true background level. Is that correct? 

If something else is stored in BKGLEVEL, it might be useful to clarify.

Reply:
I believe that the BKGLEVEL is the saved in the table above. Maybe adding new keyword saving the method used could help to remember what BKGLEVEL refers to.

Reply:
Yes, the values listed in the table are the ones that would end up being stored in the BKGLEVEL keyword for each image. Of course for a mode like "match", that value is not the true total background in each image, it's only the value necessary to get them all to match one another. They could all still have some (nearly) common residual background signal in them. So reporting, in another keyword, the bkg method that was used could be very useful, so that folks understand exactly what the values in BKGLEVEL mean. Perhaps a new keyword like "BKGMETH". It would not be possible to store the method used in the comment portion of the BKGLEVEL keyword, because the comments come from a single, static definition of the keyword.

Reply:
OK, I understand how comments work. I agree that adding a new keyword would help to keep track of things.

Reply:
I will add the BKGMETH keyword via another PR. I've included mention of it in the latest commits to the docs here.